concurrent/actor/Actor.java

```diff
@@ -25,9 +28,11 @@
 @ThreadSafe
 public abstract class Actor<ACTOR extends Actor<ACTOR>> {
 
+    private static final Logger LOG = LoggerFactory.getLogger(Actor.class);
     private static final String ERROR_ACTOR_DRIVER_IS_NULL = "driver() must not be null.";
     private final Driver<ACTOR> driver;
     private final String name;
+    private boolean terminated;
```

Comment:
This is abstracted in the next PR and as you say should be a message type of it's own

Reply:
I have not, we'll table that for discussion

Reply:
Ah in fact I misread this - it's a great catch as I thought I was only changing `reasoner/resolution/framework`. The reason for this change is to abstract termination behaviour between the resolvers and other actors, which for now is only the `Materialiser`. I've now made a `ReasonerActor` class to encapsulate common reasoner actor behaviours, anticipating that there are more actor types to come in this refactor.